src/main/java/com/sproutsocial/nsq/Publisher.java

```diff
@@ -107,12 +81,12 @@ public synchronized void publishDeferred(String topic, byte[] data, long delay,
         checkArgument(data.length > 0);
         checkArgument(delay > 0);
         checkNotNull(unit);
+        ConnectionDetails connection = balanceStrategy.getConnectionDetails();
         try {
-            checkConnection();
-            con.publishDeferred(topic, data, unit.toMillis(delay));
-        }
-        catch (Exception e) {
-            //deferred publish never fails over
+            connection.getCon().publishDeferred(topic, data, unit.toMillis(delay));
+        } catch (Exception e) {
+            connection.markFailure();
+            //deferred publish does not retry
```

Comment:
Silly question.  Does anyone know why deferred publishing doesn't retry?

Reply:
I thought the same thing. It wasn't clear to me, and I was considering adding it back in.

Reply:
Ok, I will err on the side of "keep everything the same" for now.  We can revisit it later.